text-dream/src/actions/index.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2018 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+*/
+import * as types from './types';
+
+/**
+ * Triggers the addition of a new dreaming element.
+ *
+ * @param {object} element - the element to be added to the visualization.
+ * @return {object} the action to be dispatched for this change.
+ */
+export function addDreamingElement(element) {
+  if (element.type === 'top_words') {
```

Comment:
why this special check for top_words?

Reply:
For top_words, i need a interation variable in the state, so I can use a slider etc. for going over iterations.
This variable needs to get initialized and its not really intuitive to require it in the input as it is only useful for the application to keep track of what the user wants to see.